From c46e5bc8e1612bea4e5e2bf78c20e754bed81b43 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Sat, 12 Sep 2020 15:52:47 +0200
Subject: [PATCH] ffmpeg: remove

In OpenWrt 19.07, there is a version 3.4.7, but in our feeds, we are
using 4.0.x version of ffmpeg. We want to maintain a less packages than
ever in our specific feeds and this package is updated in OpenWrt
master. Currently, there is a version 4.3.1. Let's use
for upcoming Turris OS 6.x version (I mean OpenWrt 20.xx) ffmpeg from
their feed directly.
---
 multimedia/ffmpeg/Config.in | 434 ------------------------------------
 multimedia/ffmpeg/Makefile  | 213 ------------------
 2 files changed, 647 deletions(-)
 delete mode 100644 multimedia/ffmpeg/Config.in
 delete mode 100644 multimedia/ffmpeg/Makefile

diff --git a/multimedia/ffmpeg/Config.in b/multimedia/ffmpeg/Config.in
deleted file mode 100644
index 760dad6bf..000000000
--- a/multimedia/ffmpeg/Config.in
+++ /dev/null
@@ -1,434 +0,0 @@
-menu "Configuration"
-	depends on PACKAGE_libffmpeg-custom
-
-config FFMPEG_CUSTOM_PATENTED
-	bool "Include patented codecs and technologies"
-	default BUILD_PATENTED
-
-comment "Profiles ---"
-
-config FFMPEG_CUSTOM_FFSERVER_SUPPORT
-	bool "Include support for ffserver (FFmpeg streaming server)"
-	select FFMPEG_CUSTOM_MUXER_ffm
-	select FFMPEG_CUSTOM_DEMUXER_ffm
-	select FFMPEG_CUSTOM_DEMUXER_rtsp
-	select FFMPEG_CUSTOM_PROTOCOL_rtp
-
-config FFMPEG_CUSTOM_MINIDLNA_SUPPORT
-	bool "Include support for minidlna"
-	depends on FFMPEG_CUSTOM_PATENTED
-	select FFMPEG_CUSTOM_DECODER_aac
-	select FFMPEG_CUSTOM_DECODER_ac3
-	select FFMPEG_CUSTOM_DECODER_flac
-	select FFMPEG_CUSTOM_DECODER_h264
-	select FFMPEG_CUSTOM_DECODER_jpegls
-	select FFMPEG_CUSTOM_DECODER_mp3
-	select FFMPEG_CUSTOM_DECODER_mpeg1video
-	select FFMPEG_CUSTOM_DECODER_mpeg2video
-	select FFMPEG_CUSTOM_DECODER_mpeg4
-	select FFMPEG_CUSTOM_DECODER_mpegvideo
-	select FFMPEG_CUSTOM_DECODER_png
-	select FFMPEG_CUSTOM_DECODER_wmav1
-	select FFMPEG_CUSTOM_DECODER_wmav2
-	select FFMPEG_CUSTOM_DEMUXER_aac
-	select FFMPEG_CUSTOM_DEMUXER_ac3
-	select FFMPEG_CUSTOM_DEMUXER_avi
-	select FFMPEG_CUSTOM_DEMUXER_flac
-	select FFMPEG_CUSTOM_DEMUXER_h264
-	select FFMPEG_CUSTOM_DEMUXER_matroska
-	select FFMPEG_CUSTOM_DEMUXER_mov
-	select FFMPEG_CUSTOM_DEMUXER_mp3
-	select FFMPEG_CUSTOM_DEMUXER_mpegts
-	select FFMPEG_CUSTOM_DEMUXER_mpegvideo
-	select FFMPEG_CUSTOM_PARSER_ac3
-	select FFMPEG_CUSTOM_PARSER_flac
-	select FFMPEG_CUSTOM_PARSER_h264
-	select FFMPEG_CUSTOM_PARSER_mpeg4video
-	select FFMPEG_CUSTOM_PARSER_mpegaudio
-	select FFMPEG_CUSTOM_PROTOCOL_file
-
-config FFMPEG_CUSTOM_AUDIO_DEC_SUPPORT
-	bool "Include support for most audio decoding requirements"
-	depends on FFMPEG_CUSTOM_PATENTED
-	select FFMPEG_CUSTOM_DECODER_aac
-	select FFMPEG_CUSTOM_DECODER_aac_latm
-	select FFMPEG_CUSTOM_DECODER_ac3
-	select FFMPEG_CUSTOM_SELECT_adpcm
-	select FFMPEG_CUSTOM_DECODER_alac
-	select FFMPEG_CUSTOM_DECODER_amrnb
-	select FFMPEG_CUSTOM_DECODER_amrwb
-	select FFMPEG_CUSTOM_DECODER_ape
-	select FFMPEG_CUSTOM_DECODER_atrac3
-	select FFMPEG_CUSTOM_DECODER_flac
-	select FFMPEG_CUSTOM_DECODER_mp2
-	select FFMPEG_CUSTOM_DECODER_mp3
-	select FFMPEG_CUSTOM_DECODER_mpc7
-	select FFMPEG_CUSTOM_DECODER_mpc8
-	select FFMPEG_CUSTOM_DECODER_opus
-	select FFMPEG_CUSTOM_DECODER_pcm_s16be
-	select FFMPEG_CUSTOM_DECODER_pcm_s16le
-	select FFMPEG_CUSTOM_SELECT_speex
-	select FFMPEG_CUSTOM_DECODER_vorbis
-	select FFMPEG_CUSTOM_DECODER_wavpack
-	select FFMPEG_CUSTOM_DECODER_wmav1
-	select FFMPEG_CUSTOM_DECODER_wmav2
-	select FFMPEG_CUSTOM_DECODER_wmalossless
-	select FFMPEG_CUSTOM_DECODER_wmapro
-	select FFMPEG_CUSTOM_DECODER_zlib
-	select FFMPEG_CUSTOM_DEMUXER_aac
-	select FFMPEG_CUSTOM_DEMUXER_ac3
-	select FFMPEG_CUSTOM_DEMUXER_aiff
-	select FFMPEG_CUSTOM_DEMUXER_amr
-	select FFMPEG_CUSTOM_DEMUXER_ape
-	select FFMPEG_CUSTOM_DEMUXER_avi
-	select FFMPEG_CUSTOM_DEMUXER_flac
-	select FFMPEG_CUSTOM_DEMUXER_ffm
-	select FFMPEG_CUSTOM_DEMUXER_matroska
-	select FFMPEG_CUSTOM_DEMUXER_mp3
-	select FFMPEG_CUSTOM_DEMUXER_mov
-	select FFMPEG_CUSTOM_DEMUXER_mpc
-	select FFMPEG_CUSTOM_DEMUXER_mpc8
-	select FFMPEG_CUSTOM_DEMUXER_mpegts
-	select FFMPEG_CUSTOM_DEMUXER_ogg
-	select FFMPEG_CUSTOM_DEMUXER_rm
-	select FFMPEG_CUSTOM_DEMUXER_rtsp
-	select FFMPEG_CUSTOM_DEMUXER_rtp
-	select FFMPEG_CUSTOM_DEMUXER_sdp
-	select FFMPEG_CUSTOM_DEMUXER_wav
-	select FFMPEG_CUSTOM_DEMUXER_wv
-	select FFMPEG_CUSTOM_PARSER_aac
-	select FFMPEG_CUSTOM_PARSER_aac_latm
-	select FFMPEG_CUSTOM_PARSER_ac3
-	select FFMPEG_CUSTOM_PARSER_flac
-	select FFMPEG_CUSTOM_PARSER_mpegaudio
-	select FFMPEG_CUSTOM_PARSER_opus
-	select FFMPEG_CUSTOM_PROTOCOL_file
-	select FFMPEG_CUSTOM_PROTOCOL_http
-	select FFMPEG_CUSTOM_PROTOCOL_rtp
-	select FFMPEG_CUSTOM_PROTOCOL_tcp
-	select FFMPEG_CUSTOM_PROTOCOL_udp
-
-comment "External Libraries ---"
-
-config FFMPEG_CUSTOM_SELECT_libopus
-	bool "Opus"
-
-config FFMPEG_CUSTOM_SELECT_speex
-	bool "Speex"
-
-comment "Encoders ---"
-
-config FFMPEG_CUSTOM_ENCODER_ac3
-	bool "AC3"
-	depends on FFMPEG_CUSTOM_PATENTED
-	select FFMPEG_CUSTOM_PARSER_ac3
-
-config FFMPEG_CUSTOM_ENCODER_jpegls
-	bool "JPEG-LS"
-
-config FFMPEG_CUSTOM_ENCODER_mpeg1video
-	bool "MPEG-1 Video"
-
-config FFMPEG_CUSTOM_ENCODER_mpeg2video
-	bool "MPEG-2 Video"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_ENCODER_mpeg4
-	bool "MPEG-4"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_ENCODER_pcm_s16be
-	bool "PCM signed 16-bit big-endian"
-
-config FFMPEG_CUSTOM_ENCODER_pcm_s16le
-	bool "PCM signed 16-bit little-endian"
-
-config FFMPEG_CUSTOM_ENCODER_png
-	bool "PNG"
-	select FFMPEG_CUSTOM_ENCODER_zlib
-
-config FFMPEG_CUSTOM_ENCODER_vorbis
-	bool "Vorbis"
-
-config FFMPEG_CUSTOM_ENCODER_zlib
-	bool "Zlib"
-
-comment "Decoders ---"
-
-config FFMPEG_CUSTOM_DECODER_aac
-	bool "AAC (Advanced Audio Coding)"
-	depends on FFMPEG_CUSTOM_PATENTED
-	select FFMPEG_CUSTOM_PARSER_aac
-
-config FFMPEG_CUSTOM_SELECT_adpcm
-	bool "ADPCM (multiple types)"
-
-config FFMPEG_CUSTOM_DECODER_ac3
-	bool "AC3"
-	depends on FFMPEG_CUSTOM_PATENTED
-	select FFMPEG_CUSTOM_PARSER_ac3
-
-config FFMPEG_CUSTOM_DECODER_alac
-	bool "ALAC"
-
-config FFMPEG_CUSTOM_DECODER_amrnb
-	bool "AMR-NB"
-	select FFMPEG_CUSTOM_DEMUXER_amr
-
-config FFMPEG_CUSTOM_DECODER_amrwb
-	bool "AMR-WB"
-	select FFMPEG_CUSTOM_DEMUXER_amr
-
-config FFMPEG_CUSTOM_DECODER_ape
-	bool "APE"
-
-config FFMPEG_CUSTOM_DECODER_atrac3
-	bool "ATRAC3"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_flac
-	bool "FLAC"
-	select FFMPEG_CUSTOM_PARSER_flac
-
-config FFMPEG_CUSTOM_DECODER_gif
-	bool "GIF"
-
-config FFMPEG_CUSTOM_DECODER_h264
-	bool "H.264"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_jpegls
-	bool "JPEG-LS"
-
-config FFMPEG_CUSTOM_DECODER_mp2
-	bool "MP2 (MPEG Audio Layer 2)"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_mp3
-	bool "MP3 (MPEG Audio Layer 2)"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_mpegvideo
-	bool "MPEG Video"
-
-config FFMPEG_CUSTOM_DECODER_mpeg1video
-	bool "MPEG-1 Video"
-
-config FFMPEG_CUSTOM_DECODER_mpeg2video
-	bool "MPEG-2 Video"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_mpeg4
-	bool "MPEG-4"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_mpc7
-	bool "Musepack 7"
-	select FFMPEG_CUSTOM_DEMUXER_mpc
-
-config FFMPEG_CUSTOM_DECODER_mpc8
-	bool "Musepack 8"
-	select FFMPEG_CUSTOM_DEMUXER_mpc8
-
-config FFMPEG_CUSTOM_DECODER_opus
-	bool "Opus"
-
-config FFMPEG_CUSTOM_DECODER_pcm_s16be
-	bool "PCM signed 16-bit big-endian"
-
-config FFMPEG_CUSTOM_DECODER_pcm_s16le
-	bool "PCM signed 16-bit little-endian"
-
-config FFMPEG_CUSTOM_DECODER_png
-	bool "PNG"
-	select FFMPEG_CUSTOM_DECODER_zlib
-
-config FFMPEG_CUSTOM_DECODER_vorbis
-	bool "Vorbis"
-	select FFMPEG_CUSTOM_DEMUXER_ogg
-
-config FFMPEG_CUSTOM_DECODER_wavpack
-	bool "Wavpack"
-
-config FFMPEG_CUSTOM_DECODER_wmav1
-	bool "WMAv1"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_wmav2
-	bool "WMAv2"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DECODER_zlib
-	bool "Zlib"
-
-comment "Muxers ---"
-
-config FFMPEG_CUSTOM_MUXER_ac3
-	bool "AC3"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_MUXER_ffm
-	bool "FFM (ffserver live feed)"
-
-config FFMPEG_CUSTOM_MUXER_h264
-	bool "H.264"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_MUXER_mp3
-	bool "MP3 (MPEG Audio Layer 3)"
-
-config FFMPEG_CUSTOM_MUXER_mp4
-	bool "MP4"
-
-config FFMPEG_CUSTOM_MUXER_mpeg1video
-	bool "MPEG-1 Video"
-
-config FFMPEG_CUSTOM_MUXER_mpeg2video
-	bool "MPEG-2 Video"
-
-config FFMPEG_CUSTOM_MUXER_mpegts
-	bool "MPEG-2 (TS)"
-
-config FFMPEG_CUSTOM_MUXER_ogg
-	bool "Ogg"
-
-config FFMPEG_CUSTOM_MUXER_rtp
-	bool "RTP"
-
-comment "Demuxers ---"
-
-config FFMPEG_CUSTOM_DEMUXER_aac
-	bool "AAC"
-
-config FFMPEG_CUSTOM_DEMUXER_avi
-	bool "AVI (Audio Video Interleave)"
-
-config FFMPEG_CUSTOM_DEMUXER_ac3
-	bool "AC3"
-
-config FFMPEG_CUSTOM_DEMUXER_amr
-	bool "AMR"
-
-config FFMPEG_CUSTOM_DEMUXER_ape
-	bool "APE"
-
-config FFMPEG_CUSTOM_DEMUXER_flac
-	bool "FLAC"
-
-config FFMPEG_CUSTOM_DEMUXER_ffm
-	bool "FFM (ffserver live feed)"
-
-config FFMPEG_CUSTOM_DEMUXER_h264
-	bool "H.264"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_DEMUXER_matroska
-	bool "Matroska (MKA,MKV)"
-	select FFMPEG_CUSTOM_DECODER_zlib
-
-config FFMPEG_CUSTOM_DEMUXER_mov
-	bool "MOV/MP4/M4A/3GP/3G2/MJ2"
-	select FFMPEG_CUSTOM_DECODER_zlib
-
-config FFMPEG_CUSTOM_DEMUXER_mp3
-	bool "MP3 (MPEG Audio Layer 3)"
-	select FFMPEG_CUSTOM_PARSER_mpegaudio
-
-config FFMPEG_CUSTOM_DEMUXER_mpegvideo
-	bool "MPEG Video"
-
-config FFMPEG_CUSTOM_DEMUXER_mpegps
-	bool "MPEG-2 (PS)"
-
-config FFMPEG_CUSTOM_DEMUXER_mpegts
-	bool "MPEG-2 (TS)"
-
-config FFMPEG_CUSTOM_DEMUXER_mpc
-	bool "Musepack"
-
-config FFMPEG_CUSTOM_DEMUXER_mpc8
-	bool "Musepack 8"
-
-config FFMPEG_CUSTOM_DEMUXER_ogg
-	bool "Ogg"
-
-config FFMPEG_CUSTOM_DEMUXER_rm
-	bool "RM"
-	help
-	  RealMedia format demuxer
-
-config FFMPEG_CUSTOM_DEMUXER_rtsp
-	bool "RTSP"
-	select FFMPEG_CUSTOM_DEMUXER_rm
-	select FFMPEG_CUSTOM_DEMUXER_sdp
-	select FFMPEG_CUSTOM_DEMUXER_rtp
-
-config FFMPEG_CUSTOM_DEMUXER_sdp
-	bool "SDP"
-	select FFMPEG_CUSTOM_DEMUXER_mpegts
-
-config FFMPEG_CUSTOM_DEMUXER_rtp
-	bool "RTP"
-
-config FFMPEG_CUSTOM_DEMUXER_wav
-	bool "WAV"
-
-config FFMPEG_CUSTOM_DEMUXER_wv
-	bool "WV"
-
-comment "Parsers ---"
-
-config FFMPEG_CUSTOM_PARSER_aac
-	bool "AAC (Advanced Audio Coding)"
-	depends on FFMPEG_CUSTOM_PATENTED
-
-config FFMPEG_CUSTOM_PARSER_ac3
-	bool "AC3"
-
-config FFMPEG_CUSTOM_PARSER_flac
-	bool "FLAC"
-
-config FFMPEG_CUSTOM_PARSER_h264
-	bool "H.264"
-	depends on FFMPEG_CUSTOM_PATENTED
-	select FFMPEG_CUSTOM_DECODER_h264
-
-config FFMPEG_CUSTOM_PARSER_mpegaudio
-	bool "MPEG Audio"
-
-config FFMPEG_CUSTOM_PARSER_mpegvideo
-	bool "MPEG Video"
-
-config FFMPEG_CUSTOM_PARSER_mpeg4video
-	bool "MPEG-4 Video"
-
-config FFMPEG_CUSTOM_PARSER_opus
-	bool "Opus"
-
-comment "Protocols ---"
-
-config FFMPEG_CUSTOM_PROTOCOL_file
-	bool "file:"
-
-config FFMPEG_CUSTOM_PROTOCOL_http
-	bool "http:"
-
-config FFMPEG_CUSTOM_PROTOCOL_icecast
-	bool "icecast:"
-	select FFMPEG_CUSTOM_PROTOCOL_http
-
-config FFMPEG_CUSTOM_PROTOCOL_pipe
-	bool "pipe:"
-
-config FFMPEG_CUSTOM_PROTOCOL_rtp
-	bool "rtp:"
-	select FFMPEG_CUSTOM_PROTOCOL_udp
-
-config FFMPEG_CUSTOM_PROTOCOL_tcp
-	bool "tcp:"
-
-config FFMPEG_CUSTOM_PROTOCOL_udp
-	bool "udp:"
-
-endmenu
-
diff --git a/multimedia/ffmpeg/Makefile b/multimedia/ffmpeg/Makefile
deleted file mode 100644
index 2c2ff4c18..000000000
--- a/multimedia/ffmpeg/Makefile
+++ /dev/null
@@ -1,212 +0,0 @@
-#
-# Copyright (C) 2006-2016 OpenWrt.org
-#
-# This is free software, licensed under the GNU General Public License v2.
-# See /LICENSE for more information.
-#
-
-include $(TOPDIR)/rules.mk
-
-PKG_NAME:=ffmpeg
-PKG_VERSION:=4.2.5
-PKG_RELEASE:=1
-
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_SOURCE_URL:=https://ffmpeg.org/releases/
-PKG_HASH:=574e66a7caa4b08c1d2c0e26a79559cdd23732cbc57894bd74df3989c35594c9
-
-PKG_LICENSE:=LGPL-2.1+ GPL-2+ LGPL-3
-PKG_LICENSE_FILES:=COPYING.GPLv2 COPYING.GPLv3 COPYING.LGPLv2.1 COPYING.LGPLv3
-
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)
-
-include $(INCLUDE_DIR)/package.mk
-
-define Package/ffmpeg/Default
- TITLE:=FFmpeg
- URL:=https://www.ffmpeg.org/
-endef
-
-define Package/ffmpeg/Default/description
- FFmpeg is a a software package that can record, convert and stream digital
- audio and video in numerous formats.
-endef
-
-define Package/ffmpeg
-$(call Package/ffmpeg/Default)
- SECTION:=multimedia
- CATEGORY:=Multimedia
- TITLE+= program
- DEPENDS+= +libpthread +libffmpeg
- VARIANT:=full
-endef
-
-define Package/ffmpeg/description
-$(call Package/ffmpeg/Default/description)
- .
- This package contains the FFmpeg command line tool.
-endef
-
-
-define Package/ffprobe
-$(call Package/ffmpeg/Default)
- SECTION:=multimedia
- CATEGORY:=Multimedia
- TITLE+= CLI media identifier
- DEPENDS+= +libffmpeg
- VARIANT:=full
-endef
-
-define Package/ffprobe/description
-$(call Package/ffmpeg/Default/description)
- .
- This package contains the FFprobe command line tool.
-endef
-
-define Package/libffmpeg
-$(call Package/ffmpeg/Default)
- SECTION:=libs
- CATEGORY:=Libraries
- TITLE+= libraries (full)
- DEPENDS+= +libpthread +zlib +libbz2 +libspeex +alsa-lib +LIBC_USE_MUSL:libfreetype
- PROVIDES:= libffmpeg-mini libffmpeg-full libffmpeg-custom libffmpeg-audio-dec
- VARIANT:=full
-endef
-
-define Package/libffmpeg/description
-$(call Package/ffmpeg/Default/description)
- .
- This package contains full-featured FFmpeg shared libraries.
-endef
-
-# Strip off FPU notation
-REAL_CPU_TYPE:=$(firstword $(subst +, ,$(CONFIG_CPU_TYPE)))
-# Fixup cpu types recogized by ffmpeg configure
-REAL_CPU_TYPE:=$(subst octeonplus,octeon+,$(REAL_CPU_TYPE))
-
-FFMPEG_CONFIGURE:= \
-	CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS) $(FPIC)" \
-	LDFLAGS="$(TARGET_LDFLAGS)" \
-	./configure \
-	--enable-cross-compile \
-	--cross-prefix="$(TARGET_CROSS)" \
-	--arch="$(ARCH)" \
-	$(if $(REAL_CPU_TYPE),--cpu=$(call qstrip,$(REAL_CPU_TYPE)),) \
-	--target-os=linux \
-	--prefix="/usr" \
-	--pkg-config="pkg-config" \
-	--enable-shared \
-	--enable-static \
-	--enable-pthreads \
-	--enable-zlib \
-	--disable-doc \
-	--disable-debug \
-	--enable-gpl \
-	--enable-version3 \
-	\
-	--disable-lzma \
-	--disable-vaapi \
-	--disable-vdpau \
-	--disable-outdevs \
-	--enable-avresample
-
-ifeq ($(CONFIG_SOFT_FLOAT),y)
-FFMPEG_CONFIGURE+= \
-	--disable-altivec \
-	--disable-vsx \
-	--disable-power8 \
-	--disable-armv5te \
-	--disable-armv6 \
-	--disable-armv6t2 \
-	--disable-inline-asm \
-	--disable-mipsdsp \
-	--disable-mipsdspr2 \
-	--disable-mipsfpu \
-	--disable-msa \
-	--disable-mmi \
-	--disable-fast-unaligned \
-	--disable-runtime-cpudetect
-
-else ifneq ($(findstring arm,$(CONFIG_ARCH)),)
-FFMPEG_CONFIGURE+= \
-	--disable-runtime-cpudetect
-# XXX: GitHub issue 3320 ppc cpu with fpu but no altivec (WNDR4700)
-else ifneq ($(findstring powerpc,$(CONFIG_ARCH)),)
-FFMPEG_CONFIGURE+= \
-	--disable-altivec
-endif
-
-# libavcodec/mips/aacdec_mips.c build problem
-# XXX: disable use of assembly on MIPS. Patches welcome.
-ifneq ($(findstring mips,$(CONFIG_ARCH)),)
-FFMPEG_CONFIGURE+= \
-        --disable-asm --disable-inline-asm
-endif
-
-# selectively disable optimizations according to arch/cpu type
-ifneq ($(findstring arm,$(CONFIG_ARCH)),)
-	FFMPEG_CONFIGURE+= --enable-lto
-
-	ifneq ($(findstring vfp,$(CONFIG_TARGET_OPTIMIZATION)),)
-		FFMPEG_CONFIGURE+= --enable-vfp
-	else
-		FFMPEG_CONFIGURE+= --disable-vfp
-	endif
-	ifneq ($(findstring neon,$(CONFIG_TARGET_OPTIMIZATION)),)
-		FFMPEG_CONFIGURE+= \
-			--enable-neon \
-			--enable-vfp
-	else
-		FFMPEG_CONFIGURE+= --disable-neon
-	endif
-endif
-
-ifeq ($(ARCH),x86_64)
-	FFMPEG_CONFIGURE+= --enable-lto
-endif
-
-ifneq ($(CONFIG_TARGET_x86),)
-ifeq ($(CONFIG_NASM),y)
-  FFMPEG_CONFIGURE += --enable-x86asm
-else
-  FFMPEG_CONFIGURE += --disable-x86asm
-endif
-endif
-
-define Build/Configure
-	( cd $(PKG_BUILD_DIR); $(FFMPEG_CONFIGURE) )
-endef
-
-define Build/Compile
-	$(MAKE) -C $(PKG_BUILD_DIR) \
-		DESTDIR="$(PKG_INSTALL_DIR)" \
-		all install
-endef
-
-define Build/InstallDev
-	$(INSTALL_DIR) $(1)/usr/include
-	$(CP) $(PKG_INSTALL_DIR)/usr/include/lib{avcodec,avdevice,avfilter,avformat,avresample,avutil,postproc,swresample,swscale} $(1)/usr/include/
-	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib{avcodec,avdevice,avfilter,avformat,avresample,avutil,postproc,swresample,swscale}.{a,so*} $(1)/usr/lib/
-	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/lib{avcodec,avdevice,avfilter,avformat,avresample,avutil,postproc,swresample,swscale}.pc $(1)/usr/lib/pkgconfig/
-endef
-
-define Package/ffmpeg/install
-	$(INSTALL_DIR) $(1)/usr/bin
-	$(CP) $(PKG_INSTALL_DIR)/usr/bin/ffmpeg $(1)/usr/bin/
-endef
-
-define Package/ffprobe/install
-	$(INSTALL_DIR) $(1)/usr/bin
-	$(CP) $(PKG_INSTALL_DIR)/usr/bin/ffprobe $(1)/usr/bin/
-endef
-
-define Package/libffmpeg/install
-	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/lib{avcodec,avdevice,avfilter,avformat,avresample,avutil,postproc,swresample,swscale}.so.* $(1)/usr/lib/
-endef
-
-$(eval $(call BuildPackage,ffmpeg))
-$(eval $(call BuildPackage,ffprobe))
-$(eval $(call BuildPackage,libffmpeg))
-- 
2.25.1

